addons/mail/static/src/components/composer_text_input/composer_text_input.js

```diff
@@ -74,11 +74,13 @@ class ComposerTextInput extends Component {
 
     focus() {
         this._textareaRef.el.focus();
+        this._onFocusinTextarea();
     }
 
     focusout() {
         this.saveStateInStore();
         this._textareaRef.el.blur();
+        this._onFocusoutTextarea();
```

Comment:
Hello, did you add these 2 lines since the last time I reviewed?

And if so why?

Those handlers should already be called from the DOM event when focus changes.

Reply:
Hello,
I have checked.
we have handled focusin event and here we are using focus event on an element.
so while it was doing focus on the element, focusin event won't be triggered.
that's why I have called it manually.
Thanks

Reply:
Hello, I'm not sure I understand your explanation but I just tested and the focus events are properly triggered automatically when calling the `focus` or `blur` method, so this change should not be necessary.

I tested to revert those lines manually + your new test and both work fine, so I think this should be reverted.

Reply:
@seb-odoo Removed. :+1: